core/schema/container.graphqls

```diff
@@ -261,6 +261,29 @@ type Container {
     """
     platformVariants: [ContainerID!]
   ): Boolean!
+
+  "Retrieves this container with a registry authentication for a given address."
+  withRegistryAuth(
+    """
+    Registry's address to bind the authentication to.
+    Formatted as [host]/[user]/[repo]:[tag] (e.g. docker.io/dagger/dagger:main).
+    """
+    address: String!,
+
+    """
+    Authentication information related to the address.
+    """
+    auth: RegistryAuth!
```

Comment:
cosmetic nit: I think since this is only 2 fields, it would make sense to just inline them rather than having an input

Thoughts @sipsma @helderco ?

Reply:
Yeah I'm inclined to agree since this is already an auth-focused call

Reply:
Fixed in https://github.com/dagger/dagger/pull/4502/commits/ba9c1ceed14ef618032ecc7d2bbec1584547452d